### DISCUSSION TITLE
🚀 Feature Request: Updating dockge from within dockge

### DISCUSSION THREAD

**Discussion by BenJamesAndo** [louislam/dockge#213] [Feature Request]
The ability to click a link or button from within Dockge to initiate it updating itself to the latest version.
Portainer BE has this functionality. https://docs.portainer.io/start/upgrade#from-within-portainer

**Comment by x1ao4** [comment]
You can achieve this by putting Dockge's Docker Compose file into the stacks folder, allowing Dockge to manage itself.

**Reply by mahmoudhamadeh** [reply]
I tried this, it is not working unless I'm doing something wrong. I moved the opt/dockge folder inside opt/stacks, I can see Dockge stack now in the list. When I try to update, it is pulling the new image but failed to restart. The only way to restart it is going into the terminal under opt/stacks/dockge and run docker compose up.

![image](https://github.com/louislam/dockge/assets/26927138/0a5b59e7-5ffd-46bc-ad84-b4f3d101371f)

**Reply by x1ao4** [reply]
Is your Dockge Compose file named `/opt/stacks/dockge/compose.yaml`? When I click update within Dockge, it automatically pulls the image and restarts successfully.

![WX20240629-115407@2x](https://github.com/louislam/dockge/assets/112841659/e3d5b432-bd04-4733-a2f4-9613517d537e)

**Comment by Ashkaan** [comment]
What are the exact steps?  Here's what I did:

1. Remove the current dockge instance
2. Added compose.yaml to /opt/stacks/dockge/compose.yaml
3. docker compose up -d on that folder

Inside, I get this message:

<img width="413" alt="image" src="https://github.com/user-attachments/assets/8c7b3b80-8e62-4fcd-b3d3-a2254a77761b" />

**Reply by x1ao4** [reply]
> What are the exact steps? Here's what I did:
> 
> 1. Remove the current dockge instance
> 2. Added compose.yaml to /opt/stacks/dockge/compose.yaml
> 3. docker compose up -d on that folder
> 
> Inside, I get this message:
> 
> <img alt="image" width="413" src="https://private-user-images.githubusercontent.com/5667978/433502861-8c7b3b80-8e62-4fcd-b3d3-a2254a77761b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDQ2ODgxMDksIm5iZiI6MTc0NDY4NzgwOSwicGF0aCI6Ii81NjY3OTc4LzQzMzUwMjg2MS04YzdiM2I4MC04ZTYyLTRmY2QtYjNkMy1hMjI1NGE3Nzc2MWIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDQxNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTA0MTVUMDMzMDA5WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NzBlMjllMGM2YTRmMzhhNzMzZmUxMDg5YWI3OWQ1YjFkMmVhOTFiZDMwNGRjM2E3ZGI0OGVhMWE0NjZhZDRhNCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.6bzLy66WuiR8687lxS7jB3Mb9wTPw2zqJolACh9QdbU">

You don't need to carry out the operations in Step 1 and Step 3. It seems that you've also made an error in the operation of Step 2 (the container name appears to be incorrect).

Please refer to the section "[Can I manage existing stacks?](https://github.com/louislam/dockge?tab=readme-ov-file#can-i-manage-existing-stacks)" for instructions. Letting Dockge manage itself actually means having Dockge manage the Dockge container that wasn't created within Dockge. Just like other containers that weren't created within Dockge, there's no difference.

**Comment by Ashkaan** [comment]
Also, this happened.  I'm not sure how to clear it.  I've tried deleting the inactive ones, but that deletes BOTH dockge instances.

<img width="364" alt="image" src="https://github.com/user-attachments/assets/c07257aa-b44a-42dd-a69b-eec19fedae49" />

**Comment by Ashkaan** [comment]
Ahh!  The name matters!  That solved my issue.

Thank you so much

**Comment by lehades** [comment]
@x1ao4 @Ashkaan ¿Does this still work for you?

I just started using Dockge and i'm trying this and it doesn't seem to work: i have Dockge running and self-managing it's instance: i can stop it or restart it from the web portal.. but if i try to update it it just stops and never gets back up.

stacks folder is '/opt/stacks' and dockge folder is '/opt/stacks/dockge'

<img width="1184" height="569" alt="imagen" src="https://github.com/user-attachments/assets/27eef8a5-042b-472b-8f69-adc1cbf8ac4b" />

¿Am I missing something stupid?

**Reply by Ashkaan** [reply]
I'm not using Dockge anymore, but it always worked after fixing the naming issue.

**Reply by finder39** [reply]
@lehades @andreibrebene — I ran into the same issue and built proper self-update support in my fork. The problem is that when Dockge updates its own container, the process that initiated the update gets killed mid-operation. Docker Compose v5 made this worse by being more aggressive with container recreation.

My fork handles this with safety guards so Dockge can reliably update its own stack — it works across all my servers including agents. Full disclosure, it was vibecoded with Claude Code — I'm an iOS dev, not a Docker/TypeScript expert — but it's been solid on my setup.

More details in [my post in the general discussion](https://github.com/louislam/dockge/discussions/844#discussioncomment-12733498).

- Fork: [finder39/dockge](https://github.com/finder39/dockge)
- Docker image: `ghcr.io/finder39/dockge:release` (amd64 + arm64)
- HA integration: [finder39/ha-dockge](https://github.com/finder39/ha-dockge)

**Reply by andreibrebene** [reply]
Wonderful work, if it was faster for sure i was still on dockge but right now seeing that the project is not maintained anymore i switched to Arcane.

**Reply by finder39** [reply]
Thank you, and no worries! I built this last weekend cause I wanted the new features and to have control within Home Assistant. It worked well enough I figured I'd share it out. It's a bummer its not maintained but I' glad I could give it a little more life.

I couldn't find a HA integration for Arcane. Do you know if one exists?

**Comment by mahmoudhamadeh** [comment]
Due to the limited support, I have moved from Dockge to Arcane. I'm happy
with the change since it has been working much better for my needs.

**Comment by andreibrebene** [comment]
It will not work anymore because Docker Compose v5 is more aggresive with recreating containers. Until Docker compose v5.0.0 it worked for me too.
